specs/FileTypePolicy.md

```diff
@@ -0,0 +1,222 @@
+File Type Policy API
+===
+
+# Background
+When saving a file with original SaveFilePicker, a security alert might be
+prompted, because the browser applies the [file type policies](https://learn.microsoft.com/en-us/deployedge/microsoft-edge-security-downloads-interruptions#file-types-requiring-a-gesture)
+to protect end users. However, in a WebView2 build App, when end users try
+to save a file with a certain file extension, they usually can trust the
+host App, domain and file extension. So, we provide the App developers the
+File Type Policy API to manage the file type policies dynamically.
+
+We'd appreciate your feedback.
+
+# Description
+
+We proposed the `CoreWebView2.SaveFileSecurityCheckStarting` event. You can register
+this event to get the file path, file extension and URI source information,
+when end users try to save a file from your App. Then you can apply your own
+rules to allow save the file with, or without a default warning dialog;
+to cancel the saving; and even to create your own UI to manage runtime 
+file type policies.
+
+# Examples
+## Win32 C++ 
+This example shows suppressing file type policy, security dialog, and 
+allow to save the file directly. It also blocks saving the exe file.
+The sample code will register the event with custom rules.
+```c++
+bool ScenarioFileTypePolicyStaging::AddCustomFileTypePolicies()
+{
+    if (!m_webView2Staging25)
+        return false;
+    // Register a handler for the `SaveFileSecurityCheckStarting` event.
+    m_webView2Staging25->add_SaveFileSecurityCheckStarting(
+        Callback<ICoreWebView2StagingSaveFileSecurityCheckStartingEventHandler>(
+            [this](
```

Comment:
`this` is captured without a defined lifetime.  This should probably be an owning pointer (e.g. `std::shared_ptr`) or a weak reference.

Reply:
Less of an issue for WebView since it is single-threaded. You can unregister the event handler and be confident that there will be no "late straggler" events.

Reply:
Please make sure this follows existing pattern in sample app or let me know if the rest of the sample app does this too.